### PR TITLE
Workflow to auto-lock old closed issues and PRs

### PR DESCRIPTION
**Pull request by hamishwillee**
This adds a workflow to automatically lock issues and PRs that have been closed for more than a year. 

This should reduce the instance of "necrobumping" - adding follow-on discussions to issues that are long closed and often not relevant. Even if used in "good faith", this is not a good idea because the old issue was addressed, and the detail on a year old issue is likely to be out of date or incorrect.

The implementation is directly copied from https://github.com/dessant/lock-threads#examples and uses all defaults:
- Locks issues after 365 days of a PR or issue being inactive.

NOTE: I personally think this is a good idea, and there was some support on slack. I have not yet "tested" this.

### CODE REVIEW

**Comment by wbamberg** [comment]
I think this is a good idea, but I'm not particularly qualified to review it and can't merge it (I think), so removing my request for review. @fiji-flo , @Rumyra , wdyt?

**Comment by Rumyra** [comment]
I tell you what would be good - if this was added to [mdn/workflows](https://github.com/mdn/workflows) as a shared workflow across all repos @hamishwillee - pinging @schalkneethling so he's aware 👍

**Comment by schalkneethling** [comment]
> I tell you what would be good - if this was added to [mdn/workflows](https://github.com/mdn/workflows) as a shared workflow across all repos @hamishwillee - pinging @schalkneethling so he's aware 👍

Yes, please! 😸

**Comment by hamishwillee** [comment]
Cool. I've copied this across to https://github.com/mdn/workflows/pull/38 - do I need to do anything else?

**Comment by schalkneethling** [comment]
> Cool. I've copied this across to [mdn/workflows#38](https://github.com/mdn/workflows/pull/38) - do I need to do anything else? It would be really good if this could be tested in a safe place :-)

Nope, we can discuss it further in the related pull request.

> It would be really good if this could be tested in a safe place :-)

Indeed.

File: .github/workflows/lock-closed.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: 'Lock old issues and pull requests'
+on:
+  schedule:
+    - cron: '49 11,23 * * *'
```
Comment (hamishwillee) [inline .github/workflows/lock-closed.yml:4]: @schalkneethling Updated, but I **really** don't know what I am doing here. In honesty, I'd hoped that if I found a workflow that did this, someone with more expertise would roll it out. 

Anyway, I copied the same schedule as the [idle workflow](https://github.com/mdn/content/blob/main/.github/workflows/idle-issues.yml). So this should run whenever that does, with hard coded values for locking to happen one year after closure.

**Comment by schalkneethling** [comment]
Hey there @hamishwillee! Apologies for the silence here. I am making some updates to the workflows in `mdn/workflows` so they only run on the, in most cases, origin repo and not on all forks. I will also test this one out on one of our smaller repos and get back to you here.

**Comment by hamishwillee** [comment]
Thanks! Let me know if you need me to do anything here. Github actions is not one of my strengths so I am hoping you don't need me :-0

**Comment by schalkneethling** [comment]
> Thanks! Let me know if you need me to do anything here. Github actions is not one of my strengths so I am hoping you don't need me :-0

Will do, thank you @hamishwillee

**Review by schalkneethling — APPROVED** [review]
Thanks @hamishwillee, let’s take this for a spin :)

**Comment by hamishwillee** [comment]
Thanks @schalkneethling . I was kind of expecting a rush of auto-locks but no notifications yet.

**Comment by schalkneethling** [comment]
> Thanks @schalkneethling . I was kind of expecting a rush of auto-locks but no notifications yet.

Yup, it failed :) I have logged the bug and will take care of it today.

**Pull request merged**